gpic/gitlab.go

```diff
@@ -0,0 +1,147 @@
+package gpic
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+const GL_HOSTNAME = "https://gitlab.com"
+const GL_MAX_SIZE = 200
+
+type gitlabAvatar struct {
+	ID   int64
+	size int16
+}
+
+func (this *gitlabAvatar) IsDefault() bool {
+	return false
+}
+
+// Set a default image is a custom one isn't available. GitLab doesn't yet
+// support this.
+func (this *gitlabAvatar) SetDefault(urlS string) error {
+
+	return nil
+}
+
+func (this *gitlabAvatar) setDisableDefault(disable bool) {
+}
+
+// Set the size of the avatar. A size of '0' is an alias to set maxSize.
+// Ned to edit the to only 
+func (this *gitlabAvatar) SetSize(size int16) error {
+
+	if size == -1 || size > GL_MAX_SIZE {
+		size = GL_MAX_SIZE
+	} else if size < -1 {
+		return errors.New("Size cannot be a negative number under -1.")
+	} else if size == 0 {
+		size = 0
+	} else if size <= 16 {
+		size = 16
+	} else if size <= 24 {
+		size = 24
+	} else if size <= 32 {
+		size = 32
+	} else if size <= 48 {
+		size = 48
+	} else if size <= 64 {
+		size = 64
+	} else if size <= 96 {
+		size = 96
+	} else {
+		size = GL_MAX_SIZE
+	}
+
+	this.size = size
+
+	return nil
+}
+
+func (this *gitlabAvatar) URL() (*url.URL, error) {
+
+	path := fmt.Sprintf("/uploads/-/system/user/avatar/%d/avatar.png", this.ID)
+
+	v := url.Values{}
+	if this.size != 0 {
+		path = path + "?" //Move this if other query params are added
```

Comment:
This could just be left on line 68. No need to make it optional. Again, this is fine to stay though.

Reply:
Sounds good. There was a problem when I had it in line 68 so I moved it here.